index.js

```diff
@@ -0,0 +1,74 @@
+let form = document.getElementById("new-task-form");
+let input = document.getElementById("new-task-input");
+let message = document.getElementById("message");
+let posts = document.getElementById("tasks");
+
+form.addEventListener("submit", (e) => {
+  e.preventDefault();
+  formValidation();
+});
+
+let formValidation = () => {
+  if (input.value === "") {
+    message.innerHTML = "Input field can not be blank...";
+  } else {
+    message.innerHTML = "";
+    createPostAndStoreData();
+  }
+};
+let data = [];
+
+let createPostAndStoreData = () => {
+  data.push({
+    text: input.value,
+  });
+  localStorage.setItem("data", JSON.stringify(data));
+  displayHtml();
+};
+let displayHtml = () => {
+  posts.innerHTML = "";
+  data.map(
+    (task, index) =>
+      (posts.innerHTML += `
+      <div class="task" ${index}>
```

Comment:
you can use 'index' unlimited number of times, each task should have it's key . Id is mostly used as selector. 
<div class="task" key=${index}> .  
if you just leave <div class="task" ${index}> it will parse to something like <div class="task" 1> in first iteration of data. And "1" isn\t valid html attribute. You should inspect your code using devtools, to be able to better understand this.